### ISSUE TITLE
FP for `suspicious_operation_groupings`

### ISSUE THREAD

**Issue by mi2ebi** [rust-lang/rust-clippy#17143]
### Summary

in the reproducer: i would think immediately checking `x != y` would be sufficient reason to think the choice of `f(x) && g(y)` is intentional

### Lint Name

suspicious_operation_groupings

### Reproducer

I tried this code:

```rust
#![warn(clippy::suspicious_operation_groupings)]
fn f(_: char) -> bool {true}
fn g(_: char) -> bool {true}
fn h() {
    let x = 'b'; let y = 'r';
    if f(x) && g(y) && x != y {}
}
```

I saw this happen:

```
warning: this sequence of operators looks suspiciously like a bug
 --> src/lib.rs:6:8
  |
6 |     if f(x) && g(y) && x != y {}
  |        ^^^^^^^^^^^^ help: did you mean: `f(x) && g(x)`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.96.0/index.html#suspicious_operation_groupings
note: the lint level is defined here
 --> src/lib.rs:1:9
  |
1 | #![warn(clippy::suspicious_operation_groupings)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I expected to see this happen: no lint

### Version

```text
rustc 1.96.0 (ac68faa20 2026-05-25)
binary: rustc
commit-hash: ac68faa20c58cbccd01ee7208bf3b6e93a7d7f96
commit-date: 2026-05-25
host: x86_64-unknown-linux-gnu
release: 1.96.0
LLVM version: 22.1.2
```

**Comment by mi2ebi** [comment]
i've just run into this again
